### PR TITLE
Neck Control Updates

### PR DESCRIPTION
**Pull request by wxmerkt**
- Removed neck pitch from Multisense Panel
- Added the neck pitch setting for Atlas to the Teleop Panel as well, changed Valkyrie's neck command channel
- Changed message type for DESIRED_NECK_ANGLES to joint_angles_t

### CODE REVIEW

File: src/python/director/teleoppanel.py [hunk 1]
```diff
@@ -1134,17 +1134,23 @@ def buildTabWidget(self, jointGroups):
...
-                    msg.joint_position = [0] * len(joints)
-                    for i, jointName in enumerate(joints):
-                        jointIndex = self.toJointIndex(jointName)
-                        msg.joint_position[i] = self.getJointValue(jointIndex)
-
-                    lcmUtils.publish("JOINT_POSITION_GOAL", msg)
+                    if 'neck_ay' in joints:  # Atlas
+                        neckPitchDegrees = self.getJointValue(self.toJointIndex('neck_ay'))
+                        msg = lcmdrc.neck_pitch_t()
+                        msg.utime = 0
```
Comment (patmarion) [inline src/python/director/teleoppanel.py]: i know this field wasn't set in the old code, but it should be set

File: src/python/director/teleoppanel.py [hunk 2]
```diff
@@ -1134,17 +1134,23 @@ def buildTabWidget(self, jointGroups):
...
-                    msg.joint_velocity = [0] * len(joints)
-                    msg.joint_effort = [0] * len(joints)
-                    msg.joint_position = [0] * len(joints)
-                    for i, jointName in enumerate(joints):
-                        jointIndex = self.toJointIndex(jointName)
-                        msg.joint_position[i] = self.getJointValue(jointIndex)
-
-                    lcmUtils.publish("JOINT_POSITION_GOAL", msg)
+                    if 'neck_ay' in joints:  # Atlas
+                        neckPitchDegrees = self.getJointValue(self.toJointIndex('neck_ay'))
```
Comment (patmarion) [inline src/python/director/teleoppanel.py]: variable name is wrong, it is not degrees here, which is ok, the message takes radians

File: src/python/director/teleoppanel.py [hunk 3]
```diff
@@ -1134,17 +1134,23 @@ def buildTabWidget(self, jointGroups):
...
+                        msg = lcmdrc.neck_pitch_t()
+                        msg.utime = 0
+                        msg.pitch = neckPitchDegrees
+
+                        lcmUtils.publish('DESIRED_NECK_PITCH', msg)
+                    elif 'lowerNeckPitch' in joints:  # Valkyrie
+                        msg = lcmbotcore.joint_angles_t()
+                        msg.num_joints = len(joints)
+                        msg.joint_name = joints
+                        msg.joint_position = [0] * len(joints)
```
Comment (patmarion) [inline src/python/director/teleoppanel.py]: i think this message has a utime field that you could fill in

**Comment by patmarion** [comment]
should we consolidate so that the atlas driver can receive joint_angles_t on the channel DESIRED_NECK_ANGLES?

Also, maybe you haven't used the feature before, but there is a keyboard shortcut to set neck presets, such as looking forward and looking down.  Those shortcuts are activated by pressing N+0, N+1, etc, and handled by the NeckDriver class in perception.py.  That class isn't very fancy, but if it's useful on Valkyrie, we could try to reuse it.

**Comment by patmarion** [comment]
for utime field, you can do this:

from director.utime import getUtime


msg.utime = getUtime()

**Comment by wxmerkt** [comment]
Thanks, I've just added the utime setting this as you were commenting :)
Thanks for the review - I will also take a look at the Atlas driver and panels to change it there as well - I presume there are no backwards compatibility issues with changing the message type and channel?

**Comment by mauricefallon** [comment]
> should we consolidate so that the atlas driver can receive joint_angles_t on the channel 
> DESIRED_NECK_ANGLES?

+1

**Comment by wxmerkt** [comment]
Updated this PR; now dependent on https://github.com/openhumanoids/atlas-drivers/pull/4.

**Comment by patmarion** [comment]
looks good to me

**Comment by patmarion** [comment]
please go ahead and merge when you're ready

**Pull request merged**